angular/projects/admin-nrpti/src/app/records/orders/order-add-edit/order-add-edit.component.ts

```diff
@@ -147,33 +184,49 @@ export class OrderAddEditComponent implements OnInit, OnDestroy {
       summary: this.myForm.controls.nrcedSummary.value
     };
     this.nrcedPublishStatus === 'Publish' && (order.OrderNRCED['addRole'] = 'public');
+    this.isEditing && this.nrcedPublishStatus === 'Unpublish' && (order.OrderNRCED['removeRole'] = 'public');
 
     order.OrderLNG = {
       description: this.myForm.controls.lngDescription.value
     };
     this.lngPublishStatus === 'Publish' && (order.OrderLNG['addRole'] = 'public');
+    this.isEditing && this.lngPublishStatus === 'Unpublish' && (order.OrderLNG['removeRole'] = 'public');
```

Comment:
Similar to my other comment, and maybe this is subjective, but I think its a bit awkward to set a value in the middle of a long chain like this, especially because at first glance you don't really notice it is setting a value, until you realize its a single equal and not a double or triple.

I think it reads a lot cleaner the traditional way:
```
if(this.isEditing && (this.lngPublishStatus === 'Unpublish')) {
  order.OrderLNG['removeRole'] = 'public';
}
```